### PR TITLE
add session resume feature

### PR DESCRIPTION
**Pull request by RussellChamp**
* keep track of the last route the user has visited and the time they visited
* the first time we arrive at the default home page (which is where the post-login process will drop a user) check if we should resume the prior "session"

### CODE REVIEW

File: src/authn/session.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const getLastPath = (): { path: string; lastSet: number } => {
+  return {
+    path: localStorage.getItem('last-path') || '',
+    lastSet: Number(localStorage.getItem('last-path-set')),
+  }
+}
+
+const lastPathBlacklist = [routes.ROOT, routes.HOME, routes.LOGOUT, routes.LOGGEDOUT]
+export const setLastPath = (path: string): void => {
```
Comment (RussellChamp) [inline src/authn/session.ts:14]: we do NOT want to "resume" to certain routes but we DO want to track that an attempt was made.
say the user was on the settings page for 3 hours. when they come back to that same session, they will be logged out but we will want to resume this "stale" session

File: src/pages/home.svelte [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
 import { roleSelection } from 'data/role-policy-selection'
 
-$: sendToRoleHome($roleSelection)
+$: {
+  const lastPath = getLastPath()
+  // if the last path was set withing the last 5 minutes assume we want to resume the session
+  // only ever redirect to the last path once per session
+  const shouldResumeSession = lastPath && !$initialized && isBefore(Date.now(), add(lastPath.lastSet, { minutes: 5 }))
+  if (shouldResumeSession) {
+    initialized.set(true)
```
Comment (hobbitronics) [inline src/pages/home.svelte:15]:
```suggestion
    $initialized = true
```

**Review by hobbitronics — CHANGES_REQUESTED** [review]
interesting, just one change requested.

**Review by hobbitronics — APPROVED** [review]
Nice

**Pull request merged**